Dining/views.py

```diff
@@ -310,50 +309,76 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['tab'] = "list"
 
-        context['can_delete_some'] = False
         context['entries'] = self.dining_list.dining_entries.all()
 
+        # determine whether the user has external entries added that he/she can remove until closing time
+        context['can_delete_some'] = \
+            self.dining_list.external_dining_entries().filter(user=self.request.user).count() > 0
         context['can_delete_some'] = context['can_delete_some'] * context['is_open']
+
         context['can_edit_stats'] = (self.request.user == self.dining_list.claimed_by)
         context['can_delete_all'] = (self.request.user == self.dining_list.claimed_by)
         purchaser = self.dining_list.purchaser
         context['can_edit_pay'] = (self.request.user == purchaser or
-                                        (purchaser is None and self.request.user == self.dining_list.claimed_by))
+                                   (purchaser is None and self.request.user == self.dining_list.claimed_by))
         return context
 
     def post(self, request, *args, **kwargs):
         can_adjust_stats = request.user == self.dining_list.claimed_by
         can_adjust_paid = request.user == self.dining_list.get_purchaser()
 
-        entries = {}
-        # Loop over all user entries, and store them
-        for entry in self.dining_list.dining_entries.all():
-            if can_adjust_stats:
+        if not (can_adjust_stats or can_adjust_paid):
+            messages.add_message(request, messages.ERROR, "You did not have the rights to adjust anything")
+            return HttpResponseRedirect(self.reverse('slot_list'))
```

Comment:
Return something like `HttpResponseForbidden` for invalid actions that normal users won't do because the action is not presented to them.